extensions/azurecore/src/azurecore.d.ts

```diff
@@ -75,10 +82,18 @@ declare module 'azurecore' {
 		getStorageAccounts(account: azdata.Account, subscriptions: azureResource.AzureResourceSubscription[], ignoreErrors?: boolean): Promise<GetStorageAccountResult>;
 		getBlobContainers(account: azdata.Account, subscription: azureResource.AzureResourceSubscription, storageAccount: azureResource.AzureGraphResource, ignoreErrors?: boolean): Promise<GetBlobContainersResult>;
 		getFileShares(account: azdata.Account, subscription: azureResource.AzureResourceSubscription, storageAccount: azureResource.AzureGraphResource, ignoreErrors?: boolean): Promise<GetFileSharesResult>;
-		getMigrationController(account: azdata.Account, subscription: azureResource.AzureResourceSubscription, resourceGroupName: string, regionName: string, controllerName: string, ignoreErrors?: boolean): Promise<GetMigrationControllerResult>;
-		createMigrationController(account:azdata.Account, subscription: azureResource.AzureResourceSubscription, resourceGroupName: string, regionName: string, controllerName: string, ignoreErrors?:boolean): Promise<CreateMigrationControllerResult>;
-		getMigrationControllerAuthKeys(account: azdata.Account, subscription: azureResource.AzureResourceSubscription, resourceGroupName: string, regionName: string, controllerName: string, ignoreErrors?: boolean): Promise<GetMigrationControllerAuthKeysResult>;
-
+		/**
+		 * Makes Azure REST requests to create, retireve, update or delete access to azure service's resources.
+		 * For reference to different service URLs, See https://docs.microsoft.com/en-us/rest/api/?view=Azure
+		 * @param account The azure account used to acquire access token
+		 * @param subscription The subscription under azure account where the service will perform operations.
+		 * @param serviceUrl The endpoint for the service. See https://docs.microsoft.com/en-us/rest/api/?view=Azure
+		 * @param requestType Http request method. Currently GET, PUT, POST and DELETE methods are supported.
+		 * @param requestBody Optional request body to be used in PUT and POST requests.
+		 * @param ignoreErrors When this flag is set the method will not throw any runtime or service errors and will return the errors in errors array.
+		 * @param overrideHost Set this flag when you are passing host as a part of serviceUrl. Default host is 'https://management.azure.com'
```

Comment:
I'd rather we do something like have a param that's either one of the enum values or a string if the caller wants to specify the endpoint themselves. That would make it easier for most callers to use this.

Reply:
yeah, please review it and if it is possible to extract out the some fixed pattern from the url you are expecting , I think what Charles said makes sense.